packages/flutter/lib/src/material/data_table.dart

```diff
@@ -746,12 +761,18 @@ class DataTable extends StatelessWidget {
     final double effectiveHorizontalMargin = horizontalMargin
       ?? themeData.dataTableTheme.horizontalMargin
       ?? _horizontalMargin;
+    final double effectiveCheckboxHorizontalMarginStart = checkboxHorizontalMargin
+      ?? themeData.dataTableTheme.checkboxHorizontalMargin
+      ?? effectiveHorizontalMargin;
+    final double effectiveCheckboxHorizontalMarginEnd = checkboxHorizontalMargin
+      ?? themeData.dataTableTheme.checkboxHorizontalMargin
+      ?? effectiveHorizontalMargin / 2.0;
```

Comment:
Why does this default to `effectiveHorizontalMargin / 2`? The API docs imply that it will default to 24, not 12 right?

Reply:
It is not obvious, but the end margin of the checkbox is made up of effectiveCheckboxHorizontalMarginEnd (i.e. effectiveHorizontalMargin / 2.0 ) here and the other part here:
https://github.com/flutter/flutter/blob/e030ddfee8a62116c91c35c2339deee41410f243/packages/flutter/lib/src/material/data_table.dart#L1037

It ends up being the full effectiveHorizontalMargin.

Reply:
> It is not obvious, but the end margin of the checkbox is made up of effectiveCheckboxHorizontalMarginEnd (i.e. effectiveHorizontalMargin / 2.0 ) here and the other part here:
> https://github.com/flutter/flutter/blob/e030ddfee8a62116c91c35c2339deee41410f243/packages/flutter/lib/src/material/data_table.dart#L1037
> 
> It ends up being the full effectiveHorizontalMargin.

Yes that is indeed the case :smile: